tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -513,6 +517,74 @@ def filter_yaml(key, value):
     return {node[0]: node[1] for node in (filter_yaml(k, v) for k, v in yml.items()) if node is not None}
 
 
+def expand_matrix_jobs(yml: dict) -> dict:
+    """
+    Will expand matrix jobs into multiple jobs
+    """
+
+    def get_combinated_variables(arg: dict[str, (str | list[str])]):
```

Comment:
🥜 For readability I'm not fan of method declaration like this, I prefer it aside (and below the parent method)